data_steward/tools/load_submission.py

```diff
@@ -0,0 +1,124 @@
+"""
+Loads a submission into bq from the archive
+"""
+import logging
+import argparse
+
+from google.cloud.bigquery import LoadJobConfig, Table
+from google.cloud.storage import Client
+
+from common import AOU_REQUIRED, JINJA_ENV
+from utils import pipeline_logging
+from utils.bq import get_table_schema, get_client
+import constants.bq_utils as bq_consts
+
+LOGGER = logging.getLogger(__name__)
+
+BUCKET_NAME_QUERY = JINJA_ENV.from_string("""
+SELECT bucket_name
+FROM `{{project}}.{{dataset}}.{{bucket_names_table}}`
+WHERE hpo_id = '{{hpo_id}}'
+""")
+
+
+def get_bucket(client, hpo_id):
+    bucket_name_query = BUCKET_NAME_QUERY.render(
+        project=client.project,
+        dataset=bq_consts.LOOKUP_TABLES_DATASET_ID,
+        bucket_names_table=bq_consts.HPO_ID_BUCKET_NAME_TABLE_ID,
+        hpo_id=hpo_id.upper())
+
+    bucket_name_result = client.query(bucket_name_query).result()
+    bucket_name = bucket_name_result.to_dataframe()['bucket_name'].to_list()[0]
+    return bucket_name
+
+
+def _filename_to_table_name(filename: str) -> str:
+    return filename.split('/')[-1].replace('.csv', '').lower()
+
+
+def load_stage(dst_dataset, bq_client, bucket_name, prefix, gcs_client, hpo_id):
+    """
+    Stage files from a bucket to a dataset
+
+    :param dst_dataset: reference to destination dataset object
+    :param bq_client: a BigQuery client object
+    :param bucket_name: the location in GCS containing the vocabulary files
+    :param gcs_client: a Cloud Storage client object
+    :return: list of completed load jobs
+    """
+    blobs = list(gcs_client.list_blobs(bucket_name, prefix=prefix))
+
+    load_jobs = []
+    for blob in blobs:
+        table_name = _filename_to_table_name(blob.name)
+        if table_name not in AOU_REQUIRED:
+            continue
+        schema = get_table_schema(table_name)
+        hpo_table_name = f'{hpo_id}_{_filename_to_table_name(blob.name)}'
+        fq_hpo_table = f'{bq_client.project}.{dst_dataset}.{hpo_table_name}'
+        destination = Table(fq_hpo_table, schema=schema)
+        destination = bq_client.create_table(destination, exists_ok=True)
```

Comment:
What is your rationale for `exists_ok=True` here?  I would think we wouldn't want to overwrite an existing table if we are reloading from the archive.  I would expect an error to be thrown to prevent this.

Reply:
I think by default it [appends](https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#jobconfigurationload) but you're right in that this wouldn't help us. I'll change it to throw an error instead.